src/main/java/com/pubnub/api/endpoints/Endpoint.java

```diff
@@ -393,9 +393,7 @@ protected Map<String, String> createBaseParams() {
 
     protected Map<String, String> encodeParams(Map<String, String> params) {
         Map<String, String> encodedParams = new HashMap<>(params);
-        if (encodedParams.containsKey(PubNubUtil.AUTH_QUERY_PARAM_NAME)) {
-            encodedParams.put(PubNubUtil.AUTH_QUERY_PARAM_NAME, PubNubUtil.urlEncode(encodedParams.get(PubNubUtil.AUTH_QUERY_PARAM_NAME)));
-        }
+        encodedParams.computeIfPresent(PubNubUtil.AUTH_QUERY_PARAM_NAME, (key, val) -> PubNubUtil.urlEncode(encodedParams.get(PubNubUtil.AUTH_QUERY_PARAM_NAME)));
```

Comment:
Question is what versions of Android Gradle Plugin do we support. If it's 4.0.0 and up than I guess it's fine. If something lower this change will break applications

Reply:
For reference. Keith agreed that we can go with Android Gradle Plugin 4.0.0 or higher so I will leave this as is and start using java 8 features :)